src/platformio.ts

```diff
@@ -100,6 +120,37 @@ export class PlatformIO {
         this._pioTerminal.onDidCloseTerminal(closedTerminal);
     }
 
+    private configureAutoUpdateIncludes(): void {
+        const autoUpdateIncludes = vscode.workspace.getConfiguration("platformio").get<boolean>("autoUpdateIncludes");
+        if (autoUpdateIncludes && !this._autoUpdateIncludesEnabled) {
+            const fileSystemWatcher: vscode.FileSystemWatcher = vscode.workspace.createFileSystemWatcher("**");
+            this._autoUpdateIncludesDisposables.push(fileSystemWatcher);
+            fileSystemWatcher.onDidCreate((e: vscode.Uri) => this.fileSystemDidChange(e), this, this._autoUpdateIncludesDisposables);
+            fileSystemWatcher.onDidDelete((e: vscode.Uri) => this.fileSystemDidChange(e), this, this._autoUpdateIncludesDisposables);
+            const iniFileWatcher: vscode.FileSystemWatcher = vscode.workspace.createFileSystemWatcher("**/platformio.ini");
+            this._autoUpdateIncludesDisposables.push(iniFileWatcher);
+            iniFileWatcher.onDidChange((e: vscode.Uri) => this.addIncludePath(true), this, this._autoUpdateIncludesDisposables);
+            iniFileWatcher.onDidCreate((e: vscode.Uri) => this.addIncludePath(true), this, this._autoUpdateIncludesDisposables);
+        }
+
+        if (!autoUpdateIncludes && this._autoUpdateIncludesEnabled) {
+            this.disposeAutoUpdateIncludes();
+        }
+
+        this._autoUpdateIncludesEnabled = autoUpdateIncludes;
+    }
+
+    private fileSystemDidChange(e: vscode.Uri) {
+        clearTimeout(this._debounceTimer);
+        this._debounceTimer = setTimeout(() => this.addIncludePath(true), 5000);
+    }

```

Comment:
This is not enough. If I have a test.ino file, when I run `platformio run --target idedata`, a temporary test.ino.cpp will be created then deleted. This will lead to infinite loop to call `addIncludePath`